src/main/java/org/sagebionetworks/bridge/exporter/handler/SynapseExportHandler.java

```diff
@@ -0,0 +1,229 @@
+ * will create it.
+ */
+public abstract class SynapseExportHandler extends ExportHandler {
+    private static final Logger LOG = LoggerFactory.getLogger(SynapseExportHandler.class);
+
+    // Access types are package-scoped so the unit tests can test against these.
+    static final Set<ACCESS_TYPE> ACCESS_TYPE_ALL = ImmutableSet.copyOf(ACCESS_TYPE.values());
+    static final Set<ACCESS_TYPE> ACCESS_TYPE_READ = ImmutableSet.of(ACCESS_TYPE.READ, ACCESS_TYPE.DOWNLOAD);
+
+    private static final Joiner JOINER_COLUMN_SEPARATOR = Joiner.on('\t').useForNull("");
+
+    /**
+     * Given the record (contained in the subtask), serialize the results and write to a TSV. If a TSV hasn't been
+     * created for this handler for the parent task, this will also initialize that TSV.
+     */
+    @Override
+    public void handle(ExportSubtask subtask) {
+        String tableKey = getDdbTableKeyValue();
+        ExportTask task = subtask.getParentTask();
+        Metrics metrics = task.getMetrics();
+        String recordId = subtask.getOriginalRecord().getString("id");
+
+        try {
+            // get TSV info (init if necessary)
+            TsvInfo tsvInfo = initTsvForTask(task);
+
+            // write to TSV
+            List<String> rowValueList = getTsvRowValueList(subtask);
+            tsvInfo.writeLine(JOINER_COLUMN_SEPARATOR.join(rowValueList));
+            metrics.incrementCounter(tableKey + ".lineCount");
+        } catch (BridgeExporterException | IOException | RuntimeException | SynapseException ex) {
+            metrics.incrementCounter(tableKey + ".errorCount");
+            LOG.error("Error processing record " + recordId + " for table " + tableKey + ": " + ex.getMessage(), ex);
+        }
+    }
+
+    // Gets the TSV for the task, initializing it if it hasn't been created yet
+    private TsvInfo initTsvForTask(ExportTask task) throws BridgeExporterException {
+        // check if the TSV is already saved in the task
+        TsvInfo savedTsvInfo = getTsvInfoForTask(task);
+        if (savedTsvInfo != null) {
+            return savedTsvInfo;
+        }
+
+        try {
+            // create TSV and writer
+            FileHelper fileHelper = getManager().getFileHelper();
+            File tsvFile = fileHelper.newFile(task.getTmpDir(), getDdbTableKeyValue() + ".tsv");
+            PrintWriter tsvWriter = new PrintWriter(fileHelper.getWriter(tsvFile));
+
+            // write TSV headers
+            List<String> tsvHeaderList = getTsvHeaderList();
+            tsvWriter.println(JOINER_COLUMN_SEPARATOR.join(tsvHeaderList));
+
+            // create TSV info
+            TsvInfo tsvInfo = new TsvInfo(tsvFile, tsvWriter);
+            setTsvInfoForTask(task, tsvInfo);
+            return tsvInfo;
+        } catch (FileNotFoundException ex) {
+            throw new BridgeExporterException("Error initializing TSV: " + ex.getMessage(), ex);
```

Comment:
I wonder at what point the file might throw an exception if it ever ran out of space writing to disk or we had an EBS volume error, something like that. IOException isn't declared anywhere in FileHelper methods. It might crash a thread that's processing data. What happens then?